src/waylandhelper.h

```diff
@@ -0,0 +1,38 @@
+#include <QtCore>
+
+#include <KWayland/Client/connection_thread.h>
+#include <KWayland/Client/registry.h>
+#include <KWayland/Client/surface.h>
+#include <KWayland/Client/xdgforeign.h>
```

Comment:
If you use pointers as class members - you can forward-declare them because pointers have the same size. That's what Qt do, let's do the same (note that registry cannot be forward delcared):

```cpp
#include <KWayland/Client/registry.h>

namespace KWayland::Client {
    class Surface;
    class ConnectionThread;
    class XdgExporter;
    class XdgExported;
}
class QWindow;
```

And then add the removed headers to cpp.